src/config/server.rs

```diff
@@ -183,7 +182,7 @@ impl Server {
 
     pub fn merge_right(mut self, other: Self) -> Self {
         self.apollo_tracing = other.apollo_tracing.or(self.apollo_tracing);
-        self.headers = other.headers.or(self.headers);
+        self.headers = merge_headers(self.headers, other.headers);
         self.graphiql = other.graphiql.or(self.graphiql);
```

Comment:
All the other merges seem to offload merging to their respective types. I think we should do that for headers as well and move it into `Custom`.